File: frontend/types/capacity.ts
export interface Capacity {
  totalBeds: number;
  occupiedBeds: number;
  availableBeds: number;
  criticalBeds: number;
  criticalOccupied: number;
  criticalAvailable: number;
  utilization: number;
  criticalUtilization: number;
  status: 'NORMAL' | 'WARNING' | 'CRITICAL';
  warningMessage: string | null;
}

export interface SurgeResponse {
  mode: string;
  patientsPerHour: number;
  queueLength: number;
  criticalPatients: number;
  highPatients: number;
  moderatePatients?: number;
  capacityUtilization: number;
}

File: backend/app/schemas/capacity.py
from pydantic import BaseModel
from datetime import datetime

class CapacityResponse(BaseModel):
    totalBeds: int
    occupiedBeds: int
    availableBeds: int
    criticalBeds: int
    criticalOccupied: int
    criticalAvailable: int
    utilization: float
    criticalUtilization: float
    status: str  # NORMAL, WARNING, CRITICAL
    updatedAt: datetime
